mne/io/reference.py

```diff
@@ -186,20 +186,31 @@ def add_reference_channels(inst, ref_channels, copy=True):
                         % type(inst))
     nchan = len(inst.info['ch_names'])
 
-    # "zeroth" EEG electrode dig points is reference
-    ref_dig_loc = [dl for dl in inst.info['dig'] if (
-                   dl['kind'] == FIFF.FIFFV_POINT_EEG and
-                   dl['ident'] == 0)]
-    if len(ref_channels) > 1 or len(ref_dig_loc) != len(ref_channels):
+    # only do this if we actually have digitisation points
+    if 'dig' in inst.info and inst.info['dig']:
+        # "zeroth" EEG electrode dig points is reference
+        ref_dig_loc = [dl for dl in inst.info['dig'] if (
+                       dl['kind'] == FIFF.FIFFV_POINT_EEG and
+                       dl['ident'] == 0)]
+        if len(ref_channels) > 1 or len(ref_dig_loc) != len(ref_channels):
+            ref_dig_array = np.zeros(12)
+            warn('The locations of multiple reference channels are ignored '
+                 '(set to zero).')
+        else:  # n_ref_channels == 1 and a single ref digitization exists
+            ref_dig_array = np.concatenate((ref_dig_loc[0]['r'],
+                                           ref_dig_loc[0]['r'], np.zeros(6)))
+            # Replace the (possibly new) Ref location for each channel
+            for idx in pick_types(inst.info, meg=False, eeg=True, exclude=[]):
+                inst.info['chs'][idx]['loc'][3:6] = ref_dig_loc[0]['r']
+    else:
+        # we should actually be able to do this from the montage, but
+        # it looks like the montage isn't stored, so we can't extract
+        # this information. The user will just have to call set_montag()
+        warn('No digitization found, location of new reference channel '
+             'set to zero')
```

Comment:
My impression is using digitization points at all is a minority use case for EEG people. So this might just lead to most EEG people turning off all warnings.